listenerutil/options.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package listenerutil
+
+import "net/http"
+
+// getOpts - iterate the inbound Options and return a struct
+func getOpts(opt ...Option) (*options, error) {
+	opts := getDefaultOptions()
+	for _, o := range opt {
+		if o != nil {
+			if err := o(&opts); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return &opts, nil
+}
+
+// Option - how Options are passed as arguments
+type Option func(*options) error
+
+// options = how options are represented
+type options struct {
+	withDefaultResponseHeaders    map[int]http.Header
+	withDefaultApiResponseHeaders map[int]http.Header
```

Comment:
Do we need both of these too? I'd argue we should just add what we need for now.